### ISSUE TITLE
Rediculosly low amounts of time spent cutting with an acetylene torch.

### ISSUE THREAD

**Issue by NeviNovat** [CleverRaven/Cataclysm-DDA#46257]
# Describe the bug

Cutting doors or any other things with an acetylene torch takes seconds from 2 to 15 seconds max. This is very fast. Also this can't be easy to change only the times since the amount of oxyacetylene used is directly related to the amount of time it takes for the cutting to complete. ( This was issue was mentioned in a PR see: **Additional context** section for more details ) . 

# Steps To Reproduce

1. Spawn yourself an acetylene torch.
2. Find a closed metal door.
3. Activate it and cut the door.
4. Watch and see it only takes 15 seconds.

# Expected behavior

For it to take minutes, maybe hours? I don't know how much it needs to take, if i knew I would have made a PR for it. This is how we fix this bug is to **determine the time needed** then replace the values that are wrong its simple but requires research.
Also the operation for charges consumed needs to change as well to reflect new times.

# Versions and configuration

 - OS: Windows
 - Game Version: 11280
 - Graphics version: Tiles
 - Ingame language: English
 - Mods loaded: none

# Additional context
This bug was also mentioned in "Use acetylene torch to open all locked safes #42376" but no issue was made for it, I made this issue to attract attention to it since I might not be able to fix it. Since it requires some time for research and at the moment I don't have time.
The exact code that causes this: (the screenshots where taken form a PR)
![image](https://user-images.githubusercontent.com/51816321/102916107-b0a9e800-448b-11eb-8273-057c9a09f13a.png)
![image](https://user-images.githubusercontent.com/51816321/102916024-848e6700-448b-11eb-9098-a6e0c5b9aabd.png)

**Comment by wapcaplet** [comment]
I don't know anything about oxyacetylene or metal cutting, but I think breaching a door or safe would be faster than cutting through a solid wall, if you're just cutting the deadbolt/latch mechanism or hinges. Cutting through a solid metal wall is kind of the edge case here, and should probably take significantly longer than the other metal-cutting activities, since you would need to cut a hole big enough to climb through.

**Comment by actual-nh** [comment]
Ping: @ParaplegicRacehorse - likely to take a while to answer, though...

**Comment by ParaplegicRacehorse** [comment]
Cutting through metal has a large reliance on the type of metal (it's melt/burn temperatures) and thickness of same. A gas cutting torch will rarely have a flame-tongue length of more than 6-8cm and the hottest point of that will be relatively close to the torch nozzle.

A solid metal wall is, I believe, assumed in game to be steel which will take much longer to cut through than copper, aluminum, etc. It's thickness is likely very high. I would say it is safe to assume you cannot effectively cut through a solid metal wall - hole sufficiently large to crawl through quickly - with a single large gas bottle set (ox-acet is two distinct gas bottles) or with less than most of a day. Bank vault, and other security doors, are also of metal type and thickness to make cutting through them a large expenditure of resources and time. These doors often have external fittings which could be independently cut, though, possibly speeding access to the vault. These same appendages could be got at with a hacksaw or other metal cutting tool (hmm... maybe I need to add battery operated grinders and reciprocating saws and portable bandsaws to the game.) These doors, however, also have safeguards against exactly this kind of intrusion method and are fail-safe; I think. I really don't know a whole lot about safes and security doors. Most of my metal-door expertise is water-tight hatches and fire-doors.

Self-standing discrete safes will almost universally have internal hinges, making cutting them with a torch problematic. Also, using a torch to open such a small container is likely to severely damage its contents, if not destroy them. There's just not enough internal volume to keep the heat down low enough to prevent damage to the contents.

**Comment by Fosheze** [comment]
I know a guy with an acetylene cutting torch and a lot of scrap metal up to a 1" stainless autoclave door. If someone can get me some figures on what exactly the metal doors in game are meant to be exactly, I could do some testing and get some exact numbers.

**Comment by actual-nh** [comment]
NeviNovat has pointed out that the current in-game walls can be removed with a pickaxe, which may imply something about how thin they are. Having two types of metal walls - heavy/armored and not - as NeviNovat suggested may be at least a partial solution for this.

**Comment by NeviNovat** [comment]
The walls are made from 4 steel plating ("A piece of armor plating made of steel.") as opposed to hard plating ("A piece of very thick armor plating made of steel."), if the plates are something like 1.5 cm then you could use an acetylene torch on a wall and expect reasonable results. But I do agree a wall can transfer heat better and result in more wasted oxyacetylene.

**Comment by actual-nh** [comment]
What's the volume in total? (Or the weight - density for steel, or possibly budget_steel, is in materials.json.) And what size is 1 tile in the game supposed to be?

**Comment by NeviNovat** [comment]
> 
> 
> What's the volume in total? (Or the weight - density for steel, or possibly budget_steel, is in materials.json.) And what size is 1 tile in the game supposed to be?

https://www.cdda-trunk.chezzo.com/steel_plate
For steel plating: Volume: 3 L Weight: 28.40 lbs /12.88 kg
https://www.cdda-trunk.chezzo.com/hard_plate
For hard plating: Volume: 3 L Weight: 71.96 lbs /32.64 kg

Yes volume is the same.

**Comment by actual-nh** [comment]
I have to wonder if that volume is meant to indicate the problems with carrying them, not their actual volume - if it were the actual volume, those densities are rather far apart, to put it mildly...

**Comment by NeviNovat** [comment]
> 
> 
> I have to wonder if that volume is meant to indicate the problems with carrying them, not their actual volume - if it were the actual volume, those densities are rather far apart, to put it mildly...

I might make another pull request that changes the volume based on wiki values of between 7,750 and 8,050 kg/m3. If I where to calculate the thickness I would only look at the weight and area covered anyway. ( I don't know how much they cover BTW )

**Comment by ParaplegicRacehorse** [comment]
I can personally attest to requiring around 20 minutes to cut 1/4" (5-6mm) thick soft-steel plate, 8" (10-11cm) wide into two pieces with an ox-acet torch. I cannot make a comment as to how much of the two gases was used for this same cut.

**Comment by ParaplegicRacehorse** [comment]
Originating-post specifically mentions metal doors. If the cutting is being done on the hinged side of the door, time required to remove the door is not very much longer than needed to cut the hinges. At a guess: two or three minutes per hinge. But, if you're on the hinged side, why not just knock out the hinge-pins (3-5 seconds per hinge, plus time required to ready tools) ?

This does not apply to vault doors, which are designed to be resistant to this kind of tampering.

Cutting through a wall or steel bulkhead is another matter entirely. Same with vehicular cutting or free-standing safes.

**Comment by actual-nh** [comment]
Using it to disable the lock/bolt instead - how much of a risk will there be of instead making the door unopenable?

**Comment by ParaplegicRacehorse** [comment]
Depends on the door. Secure-vault doors (think banks) have failsafes for attacks against exposed parts of mechanism.

It also occurs to me that many doors, including metal ones, can just have the knob/handle beaten off with a sledge hammer or removed with screwdriver, and the closing-bolt removed that way. Again, this only for doors not designed with failsafes against this kind of attack (ie: the vast majority of doors.)

**Comment by actual-nh** [comment]
That is certainly true for doors in which the handle's associated bolt is the only one keeping them shut.

**Comment by actual-nh** [comment]
> > I have to wonder if that volume is meant to indicate the problems with carrying them, not their actual volume - if it were the actual volume, those densities are rather far apart, to put it mildly...
> 
> I might make another pull request that changes the volume based on wiki values of between 7,750 and 8,050 kg/m3. If I where to calculate the thickness I would only look at the weight and area covered given that the volume is the same. ( I don't know how much area they cover BTW )

materials.json has "hard steel" as 40 kg/L (4000 kg/m3) and other steels, including "budget steel", at 30 kg/L (3000 kg/m3). Hrm.

> This means that steel plating has about ~1.6L of steel, and the heavy plating has ~4.1L of steel.

So 1.6L or 4.1L *4 for a wall section, or is that including the wall section multiplier?

**Comment by NeviNovat** [comment]
Metal walls are made of metal plating
![image](https://user-images.githubusercontent.com/51816321/102987161-99fea200-451a-11eb-947f-d28fac0da6ec.png)
14cm is prob the diagonal of a square plate ( that means an area of 100cm^2 ) ... or its a rectangle ... then IDK.
![image](https://user-images.githubusercontent.com/51816321/102987259-bd295180-451a-11eb-83ea-ef081cc0870e.png)
Same 14cm as above.

If we assume the plate to be square (the thickest one) then the volume is ~ all its sides times its thickness so : 
10cm * 10cm *h=1.6L => 100(cm^2)*h=1600(cm^3) => 16cm=h this means a "plate" of 10cm,10cm,16cm that's more of a cube then a plate! Clearly something is wrong with the length of the plate as well. I won't bother with the calculations for the hard plating.

> materials.json has "hard steel" as 40 kg/L (4000 kg/m3) and other steels, including "budget steel", at 30 kg/L (3000 kg/m3). Hrm. (Water's density is 1 -> 1 g/mL -> 1 kg/L.)

Did I do something wrong or are the materials.json wrong about steel? Also budget steel is steel that is not used in plating but in weapons and is supposed to be low carbon or something like that.
I mean : 
![image](https://user-images.githubusercontent.com/51816321/102988523-af74cb80-451c-11eb-917b-1a3acbe631dc.png)
Source: https://www.convertunits.com/from/kg/m3/to/kilogram/liter


> So 1.6L or 4.1L *4 for a wall section, or is that including the wall section multiplier?

A metal wall uses up 4 steel plating in total so that would be 1.6L * 4. What is a "wall section multiplier"?

**Comment by actual-nh** [comment]
("Wall section multiplier": that's the only thing I could think of calling the *4.) If it's a plate that's 100 cm (as in 1 meter) on each side, that fits with the 1m x 1m x 4m [estimate](https://github.com/CleverRaven/Cataclysm-DDA/discussions/46267#discussioncomment-236234) by @anothersimulacrum for one tile.

Looks like I slipped a decimal point on my density conversions; sorry! From looking at the git "blame" for materials.json, it appears that the "densities" in there are supposed to be kg/250 mL, so 4* the kg/L or g/cc (cc = cubic cm = mL). It is then saying that generalized "steel" is a density of 7.5 kg/L and "hard steel" is 10 kg/L. The latter is, umm, rather too high. The density for water in materials.json (1) is in error by a factor of 4, since water at its highest liquid density (4 degrees C) is 1 g/mL. (This is actually how 1 gm was originally _defined_ in the metric system, BTW.)

I suspect nobody noticed the value for water was off because of lack of use of the density field in the code, as witness `// TODO: MATERIALS add a density field to materials.json` from [item.cpp](https://github.com/CleverRaven/Cataclysm-DDA/blob/b85c4a930b74d3e9f2249c116561a811bc0a442f/src/item.cpp#L4956). When the density of water is fixed, a comment that the density is in kg per 250 mL would be very helpful in the future...

The "volume" in the game is generally how much volume is needed to store the thing, not the literal volume. 12.88 kg, at a density of 7.5 kg/L, gives a volume of roughly 1.7L, or 0.0017 m^3. Multiply by 4 for a wall section, but divide by 1m x 4m for width x z-height of one tile, gives you 0.0017 meters thick, or 1.7 mm. For the hard steel one, using your value of 8 kg/L, would be ~4.1L as you said (4.08L to be precise). That would be 0.00408 meters thick, or ~4 mm thick.

I don't know how proportional cutting speed is to thickness of the material (it might well be proportional to thickness squared or similar). If we go with the assumption/guess that the soft steel 1/4" (6.35 mm) that @ParaplegicRacehorse mentioned is about equivalent to 4 mm of "hard steel", then the cutting time is ~1 minute per cm of cut for the heavy steel wall. Cutting out a 1 m x 1 m hole in a "hard steel" wall would then take ~400 minutes, or 6 hours and 40 minutes.

**Comment by NeviNovat** [comment]
" Thicknesses can vary significantly; extremely thin sheets are considered foil or leaf, and pieces thicker than 6 mm (0.25 in) are considered plate steel or "structural steel". "
Source: https://en.wikipedia.org/wiki/Sheet_metal
So hard plating can be considered "sheet metal" if you consider the plating to be 1m x 1m.

> then the cutting time is ~1 minute per cm of cut for the heavy steel wall. 

1 minute per cm? That's super slow for only 6.35 mm. I mean this guy done it much faster for 1/2 inch ( 13mm ) steel, that's 2 times as thick (https://youtu.be/6g6g3tsHyl8?t=42)  the only thing that is interesting is that preheating is hard to do when you are working on a big plate. And there are other videos like:  https://www.youtube.com/watch?v=RcXbV0hSQDQ where they cut in a sizable plate this shows that heat dispersal isn't that big of an issue. ( but that video doesn't show their set-up and I hear pressure is an important thing )

**Comment by actual-nh** [comment]
@ParaplegicRacehorse - how carefully were you cutting the plate? It may well vary by precision (as well as other factors like whether - as with a wall - there's metal around that can conduct heat away).

**Comment by kevingranade** [comment]
Several things crop up here:
1. The rationale of "you can breach this with a pickaxe" isn't valid, that is in most cases an error itself.
2. Oxyacetelene vs iron/steel in particular is a special case, once you preheat the metal, you feed it excess preheated oxygen and the iron itself combusts to maintain the cutting temperature, so you can cut far more than the size of the torch flame would imply.  HOWEVER, the process can be interrupted by other materials, such as if you alternate iron and concrete, which AFAIK is why safes have a concrete layer.
3. "Depends on the door. Secure-vault doors (think banks) have failsafes for attacks against exposed parts of mechanism.", this is the case even with commercial metal doors, which tend to have buried hinges, multiple locks, and internal bars.
4. Trying to reason about what kind of door/wall it is based on the properties of the materials isn't a very reliable mechanism, it would be a lot more productive to try to reason about "what kind of door or wall is this" and start from there, even adjusting other stats to match.

I'd definitely agree we need more door/wall types for this, some are quite thin and can be taken apart with a crowbar, much less something heavier, and others are going to be specifically hardened against every form of attack we have in the game and would require a serious time investment to breach.
Also time to use an acetelene torch at all is pretty short, much less getting serious work done with it.

**Comment by NeviNovat** [comment]
Something about making 3 wall types was mentioned by me in "Make acetylene torch cut metal walls. #46255" as a response to someone else suggesting new walls.
I saw this as being the most optimal solution since we already have 3 types of metal construction material for the walls.
> Maybe the best solution would be making 3 walls:
> 
>    - sheet metal wall , can be bashed in but its quicker to cut with acetylene, or other power tools. ( for example: silo walls )
>    - metal wall, can't be bashed requires tools like acetylene torch or probably other tools can cut it as well. ( for example: city banks vaults, prisons maybe )
>    - hard metal wall (made out of hard plating as opposed to normal steel plating), can't be bashed , can't be penetrated with normal acetylene torch or normal tool ,maybe with industrial ones or other more advanced power tools/ explosives. ( for example: lab entrance walls, nuclear bunkers, or when you want to contain something really dangerous with 100% certainty )

But with this new information: 
> HOWEVER, the process can be interrupted by other materials, such as if you alternate iron and concrete

My suggestion is a bit wrong ( at least at the example part ) since concrete will be backing metal walls in a lot of cases, maybe adding wall types that have concrete backing along side ones that don't. Or better just think what walls would be used in what buildings and add them to the game? ( so we don't end up with wall types that are not going to be used. )
In any case this looks like a new issue about adding new metal walls. 
I might open a new issue about this when I have time, when I do I will edit stuff to mention it here.

**Comment by Emotions211** [comment]
It's obvious none of you have ever actually worked with an Oxy-fuel torch (There's more than just good old Acetylene for fuel and this does affect burn temperature, cutting rate, and cut depth). You've already done the math to figure the plate thickness used in a metal wall at about 4-6mm (Just under a quarter inch). Even for a combination welding/cutting torch that's easy to cut if you know what you're doing. Regular mixture for oxy-acetylne cutting is about 40 PSI Oxy and about 10 PSI Acetylene AT THE REGULATORS.

In operating a COMBINATION torch for cutting. These are the set up steps.

1. Ensure all valves are closed on torch body, head, manifold/cylinders and regulators are fully closed (Regulators screw towards the gauge to allow gas through, with the valve "out" there should be no pressure if the manifold is open)
2. Open Manifold/Cylinders fully
3. Open regulators to cutting mix (40Psi Oxy, 10PSi Acetylene)
4. Open the first oxy valve on the body fully
5. Holding a starter of some kind, open acetylene valve on the torch body 1 quarter turn and ignite the flame. Adjust fuel until it's clear of black smoke and strong
6. Crack open oxy valve on the torch head and slowly adjust until you achieve a neutral flame. (In this state you can begin welding thin metal or heating it to work)
7. Depress cutting handle and observe flame remains neutral with extra oxy. Adjust if needed. Once set, begin cutting.

In order to actually begin cutting, sweat the plate in-front of where you want to cut initially, this helps with actually cutting. Normally you'll start at an edge as it's faster but you can start in the middle of a plate for a more difficult start. This is what I'll explain.

1. Bring torch to point neutral flame is touching (or almost) the plate. DO NOT TAP THE PLATE, you will damage the cutting torch.
2. Heat point until a reasonably sized puddle forms, this is based on experience to know when it's ready.
3. Depress cutting lever fully and raise torch away from the plate slightly. Molten metal doesn't like having gas at 40psi being thrown at it.
4. Move torch back towards plate and begin moving in the direction you want to cut. Maintaining a constant speed, kerf lines in the cut material can tell you if you're too fast, slow, at a bad angle or wrong mixture after a cut.

That's simplified but basically it. Oxy cutting actually blows more of the material out the back instead of burning or vaporizing it. Iron can only burn in an oxygen rich environment or when a large surface area can be exposed (steel wool burns for this reason). If you're cutting a steel plate by hand, you can tell if you're cutting through by the fact you'll be spraying sparks out the back. If not, you're just vaporizing a section of plate and are either too fast or don't have the torch set properly.

Due to the fact that acetylene burns hot enough to actually vaporize steel (best shown when attempting to weld 2 1/16th plates together without a filler rod), in theory you could cut steel without a cutting attachment but it would take several hours, if not days to actually do so with a welding head.

In my personal experience. Cutting a quarter inch plate in half is about 20-30 seconds per 4 inches. Even then I'm not all that great at it.

Here's a video of a guy explaining how to cut, pierce and what going to slow or fast does. Descrption says half inch plate. Most plates like that are 4 inches wide.

https://www.youtube.com/watch?v=7EGmrPiumEU&ab_channel=freddytk421

It takes him about 15 seconds to cut a half inch plate in half (Probably 4 inches wide).

Here's another video of a man cleaning up an I-Beam cut, Best guess at dimensions is 4 wide, 6 tall and an 1/8th inch spine.

https://www.youtube.com/watch?v=Zovvbq6Qp_s&ab_channel=Weld.com

It takes him about 13 seconds to cut through the spine.

If it taking 15 seconds is a hold over from 1 turn being 6 seconds, not 1. That would still be way too fast at 90 seconds. Assuming the hinge/mounts are unreachable, you would have to cut out around the plate until it falls out. If I'm doing my math right, and read the earlier comments right as well, the player is cutting out a 3 foot by 12 foot by (aprox) 0.25 inch section in 15 seconds (assuming a door is 12 feet?). That's a total length of 30 feet in under 15 seconds. Even in that first video I linked it takes him about 15 or so seconds to create a 1/4 inch bolt hole. Assuming a constant rate of cutting (and ignoring prep-time) is about 15 seconds for a 4 inch cut on 1/4th inch plate. It would take (360 / 4 x 15} 1350 seconds (22.5 minutes). Raise it up to 30 minutes total to take down the rest of the metal and setup/packup time and we would be good assuming all walls and doors are a 360 inch perimeter (Base+heightx2)

I probably read something wrong *somewhere* since there is no way a door frame tops out at 12 feet. At around 8 feet tall it would take just over 15 minutes.

Also I'm just gonna assume we're treating walls as basically being flat planes with no internal supports. So it's just a free standing sheet of metal that is held in place by "nothing". As for that one measurement of 1.7mm, a good swing from a heavy object would dent a plate that thick if not outright tear through it, a fire-axe for example. For the sake of everyone's sanity, lets assume its .25 inch. That will stop basically any unpowered hand tools and pistol fire.

tl;dr Yes it's way too fast. It should take more than 15 seconds to cut out a metal door or wall. Closer to 20 or 30 minutes for a door would make much more sense.

**Comment by actual-nh** [comment]
Another question is how long it should take to cut through rebar - see #47701.

**Comment by kasanryukin** [comment]
> 
> 
> Another question is how long it should take to cut through rebar - see #47701.

If you're just cutting rebar up to 1/2inch in diameter it takes seconds.   I can cut up 24' long 1/4" diameter rebar into 6 sections in about a minute of actual cutting.  It takes longer to move the pieces into position than actually cutting it.
